plugins/env/decorators/project_decorator.rb

```diff
@@ -2,8 +2,11 @@
 Project.class_eval do
   include AcceptsEnvironmentVariables
 
+  has_many :external_environment_variable_groups, dependent: :destroy
   has_many :project_environment_variable_groups, dependent: :destroy
   has_many :environment_variable_groups, through: :project_environment_variable_groups, inverse_of: :projects
+  accepts_nested_attributes_for :external_environment_variable_groups, allow_destroy: true,
+    reject_if: ->(a) { a[:name].blank? && a[:url].blank? }
```

Comment:
we want to show `needs url` validation, so accept any partially filled form ?
```suggestion
    reject_if: ->(a) { a[:name].blank? || a[:url].blank? }
```